### PR TITLE
mostly pluggable models

### PR DESCRIPTION
**Pull request by cpoerschke**
(some constructors and functions marked @Deprecated and these would be removed in additional commits to this branch/pull request or in a future pull request once the (test) callers have stopped using them)

### CODE REVIEW

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java [hunk 1]
```diff
@@ -16,73 +16,217 @@
...
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.util.SolrPluginUtils;
 
 /**
  * Contains all the data needed for loading a model.
  */
 
 public abstract class LTRScoringModel {
 
+  public static class ModelFeature  implements FeatureWeightCreator {
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java]: Let's rip out the merging of normalizers + features from this pluggable scoring models pull request so we don't hold up the other tickets dependent on pluggable models with this added change.

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java [hunk 2]
```diff
@@ -16,73 +16,217 @@
...
+          className,
+          LTRScoringModel.class,
+          new String[0], // no sub packages
+          new Class[] { String.class, List.class, List.class, String.class, List.class, Map.class },
+          new Object[] { name, features, norms, featureStoreName, allFeatures, params });
+      if (params != null) {
+        SolrPluginUtils.invokeSetters(model, params.entrySet());
+      }
+    } catch (final Exception e) {
+      System.out.println("cpoerschke debug: caught "+e);
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java]: prints needed? if so switch to logger instead (for other locations too if they exist)

File: solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/FeatureWeightCreator.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.solr.ltr.ranking.Feature.FeatureWeight;
+import org.apache.solr.request.SolrQueryRequest;
+
+public interface FeatureWeightCreator {
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/java/org/apache/solr/ltr/ranking/FeatureWeightCreator.java]: This new file should be pulled out into a new PR, but if you want to keep the rename of Feature::createweight to Feature::createFeatureWeight in this PR, I'm ok with with that.

File: solr/contrib/ltr/src/test-files/modelExamples/lambdamart_model.json [hunk 4]
```diff
@@ -8,29 +8,29 @@
     "params":{
         "trees": [
             {
-                "weight" : 1,
+                "weight" : "1f",
                 "tree": {
                     "feature": "matchedTitle",
-                    "threshold": 0.5,
+                    "threshold": "0.5f",
```
Comment (mnilsson23) [inline solr/contrib/ltr/src/test-files/modelExamples/lambdamart_model.json]: Why does this no longer work if they are floats like before?

**Comment by cpoerschke** [comment]
abandoning in favour of #105

**Pull request closed without merging**